shacl/generate_shacl.py

```diff
@@ -0,0 +1,60 @@
+from rdflib import Graph, Literal, BNode, URIRef
+from rdflib.collection import Collection
+
+import sys
+sys.path.append('..')
+from bricksrc.namespaces import RDF, RDFS, BRICK, BSH, SH, SKOS
+from bricksrc.namespaces import bind_prefixes
+# add SHACL shapes from properties
+from bricksrc.properties import properties as property_definitions
+
+G = Graph()
+bind_prefixes(G)
+A = RDF.type
+
+for name, properties in property_definitions.items():
+
+    for pred, obj in properties.items():
+        sh_prop = BNode()
+
+        # We are only concerned with properties that have RDFS.domain or .range
+        # predicate.  The temporary replacements for those predicates are
+        # brick:expectedDomain and :expectedRange.  See properties.py for
+        # explanation.
+
+        if pred == BRICK.expectedRange:
+            shapename = f"{name}RangeShape"
+            G.add((BSH[shapename], SH['property'], sh_prop))
+            G.add((BSH[shapename], A, SH.NodeShape))
+
+            G.add((sh_prop, SH['path'], BRICK[name]))
+            G.add((sh_prop, SH['class'], obj))
+            G.add((BSH[shapename], SH.targetSubjectsOf, BRICK[name]))
+            G.add((sh_prop, SH['message'],
+                   Literal(f"Property {name} has object with incorrect type")))
+
+        if pred == BRICK.expectedDomain:
+            shapename = f"{name}DomainShape"
+            G.add( (BSH[shapename], A, SH.NodeShape))
+            G.add( (BSH[shapename], SH.targetSubjectsOf, BRICK[name]))
+            G.add( (BSH[shapename], SH['class'], obj))
+            G.add( (BSH[shapename], SH['message'],
```

Comment:
To keep style consistent, remove the leading space in front of the argument to `G.add()`

Reply:
done